scripts/lib.sh

```diff
@@ -16,13 +16,57 @@ configure-sentry-cli() {
     fi
 }
 
+# I'm assuming sw_vers is only available on Mac
+query-mac() {
+    require sw_vers && return 0
+    return 1
+}
+
 query_big_sur() {
     if require sw_vers && sw_vers -productVersion | grep -E "11\." > /dev/null; then
         return 0
     fi
     return 1
 }
 
+sudo-askpass() {
+  if [ -z "${sudo-askpass-x}" ]; then
+    sudo --askpass "$@"
+  else
+    sudo "$@"
+  fi
+}
+
+# After using homebrew to install docker, we need to do some magic to remove the need to interact with the GUI
+# See: https://github.com/docker/for-mac/issues/2359#issuecomment-607154849 for why we need to do things below
+init-docker() {
+  # Need to start docker if it was freshly installed (docker server is not running)
+  if query-mac && ! require docker && [ -d "/Applications/Docker.app" ]; then
+    echo "Making some changes to complete Docker initialization"
+    # allow the app to run without confirmation
+    xattr -d -r com.apple.quarantine /Applications/Docker.app
+
+    # preemptively do docker.app's setup to avoid any gui prompts
+    sudo-askpass /bin/mkdir -p /Library/PrivilegedHelperTools
+    sudo-askpass /bin/chmod 754 /Library/PrivilegedHelperTools
+    sudo-askpass /bin/cp /Applications/Docker.app/Contents/Library/LaunchServices/com.docker.vmnetd /Library/PrivilegedHelperTools/
+    sudo-askpass /bin/cp /Applications/Docker.app/Contents/Resources/com.docker.vmnetd.plist /Library/LaunchDaemons/
+    sudo-askpass /bin/chmod 544 /Library/PrivilegedHelperTools/com.docker.vmnetd
+    sudo-askpass /bin/chmod 644 /Library/LaunchDaemons/com.docker.vmnetd.plist
+    sudo-askpass /bin/launchctl load /Library/LaunchDaemons/com.docker.vmnetd.plist
+  fi
+  start-docker
+}
+
+start-docker() {
+  if query-mac && ! docker system info &>/dev/null; then
```

Comment:
Oh, this stuff / `init-docker` seems to be for CI only?

Reply:
https://github.com/getsentry/sentry/pull/24563#discussion_r604374110

I see, should probably add a comment above these functions noting it's for CI only, if that's the case.

Reply:
I've added some notes.

This is Mac specific because the execute `docker` won't even work until the app has been opened once. After that, the `docker` executable works as expected.